source/distributed/tria_base.cc

```diff
@@ -143,11 +140,31 @@ namespace parallel
   }
 
   template <int dim, int spacedim>
-  const std::vector<unsigned int> &
-  TriangulationBase<dim, spacedim>::n_locally_owned_active_cells_per_processor()
-    const
+  const std::vector<unsigned int>
+  TriangulationBase<dim, spacedim>::
+    compute_n_locally_owned_active_cells_per_processor() const
   {
-    return number_cache.n_locally_owned_active_cells;
+#ifdef DEAL_II_WITH_MPI
+    std::vector<unsigned int> n_locally_owned_active_cells_per_processor(
+      Utilities::MPI::n_mpi_processes(this->mpi_communicator), 0);
+
+    if (this->n_levels() > 0)
+      {
+        const int ierr =
+          MPI_Allgather(&number_cache.n_locally_owned_active_cells,
+                        1,
+                        MPI_UNSIGNED,
+                        n_locally_owned_active_cells_per_processor.data(),
+                        1,
+                        MPI_UNSIGNED,
+                        this->mpi_communicator);
+        AssertThrowMPI(ierr);
+      }
+
+    return n_locally_owned_active_cells_per_processor;
+#else
+    return std::vector<unsigned int>();
```

Comment:
This seems wrong -- we should return a vector of size 1 with `n_dofs()` as an element.